common/containers/Tabs/Swap/components/CurrentRates.js

```diff
@@ -1,24 +1,16 @@
 import React, { Component } from 'react';
 import translate from 'translations';
-import PropTypes from 'prop-types';
 import { toFixedIfLarger } from 'utils/formatters';
+import { Pairs } from 'actions/swapTypes';
 
 export default class CurrentRates extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      ETHBTCAmount: 1,
-      ETHREPAmount: 1,
-      BTCETHAmount: 1,
-      BTCREPAmount: 1
-    };
-  }
+  props: Pairs;
 
-  static propTypes = {
-    ETHBTC: PropTypes.number,
-    ETHREP: PropTypes.number,
-    BTCETH: PropTypes.number,
-    BTCREP: PropTypes.number
+  state = {
```

Comment:
You could loop over the pairs keys object and set `this.state[`${key}Amount`] = 1;` in the constructor. Likewise, I think the rendering could also potentially be looped over.